### PR TITLE
Package courier

### PR DESCRIPTION
**Pull request by yomashExpel**
Update the Send function signature to include a context, and strict parameters. Add a test.  Currently packaged for expel-io.

## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: send_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			assert.Equal(t, "/send", req.URL.String())
+
+			decoder := json.NewDecoder(req.Body)
+			err := decoder.Decode(&requestBody)
+			if err != nil {
+				t.Error(err)
+			}
+
+			rw.Header().Add("Content-Type", "application/json")
+			// nolint
```
Comment (DannyDouglass) [inline send_test.go:47]: @yomashExpel out of curiosity, why was lint skipped here?

File: README.md [hunk 2]
```diff
@@ -24,21 +24,20 @@ import (
...
-                        phone_number: "555-228-3890"
-                },
-        data: {} // optional variables for merging into templates
-        overrides: {} // optional http provider overrides
-        }`)
-        response, err := client.Send(message)
-        if err != nil {
-                log.Fatalln(err)
+
+        myData := struct {
```
Comment (DannyDouglass) [inline README.md:28]: small change request: `myData` => `data`. reason being is that `data` is request-scoped for recipients (not "mine").

File: README.md [hunk 3]
```diff
@@ -24,21 +24,20 @@ import (
...
+                foo string
+        }{
+                foo: "bar",
         }
-        log.Println(response.MessageId)
+        myProfile := struct {
```
Comment (DannyDouglass) [inline README.md:33]: small change request: `myProfile` => `profile`. reason being is that `profile` is request-scoped for recipients (not "mine").

File: send_test.go (hunk 1 above)
Comment (yomashExpel) [reply send_test.go:47]: it was for the ignored returned values, as I didn't need it for the purposes of the test I was writing

Comment (yomashExpel) [reply send_test.go:47]: actually, i went back to double check and its for ignoring the return error, which I shouldn't.  I will remove and handle that error

**Pull request merged**